### ISSUE TITLE
/connect/<provider> works even though provider is not setup

### ISSUE THREAD

**Issue by Lordmau5** [simov/grant#91]
So right now I only have Twitch and Patreon setup.
They both work, so that's cool - really easy setup as well, thank you for making such a great utility :)

However, when I try to access `server/connect/discord` for example, without having it setup, it still forwards to the Discord OAuth page.

It's not really a huge bug, but it still seems... redundant...

Is there any way this could be addressed?

**Comment by simov** [comment]
Put that way it sounds like a bug, though that's actually a feature, I have tests for it. It's related to the [Dynamic Overrides](https://github.com/simov/grant#dynamic-override). I'll have to think about it.

In case your Node process sits behind proxy, for example Nginx, you can easily filter out any route except the configured ones.

**Comment by Lordmau5** [comment]
Ah, Dynamic Overrides, right. I didn't think of this.

Perhaps this could be added as a config option within the `server` config block?

**Comment by simov** [comment]
Yes, a config option is the way to go, though I'm still trying to rationalize it. Either way the `/connect/:provider` route is going to be hit, the only difference would be that it will throw an error instead of trying to authorize. Still a better approach would be to handle that in the upper proxy layer, if there is one.

**Comment by simov** [comment]
Alternative approach would be to handle the provider filtering in your own app server:

```js
var express = require('express')
var session = require('express-session')
var grant = require('../../').express()

var config = require('./config.json')

express()
  // check if the provider is allowed
  .use('/connect/:provider/:override?', (req, res, next) => {
    // to make this condition dynamic you can use:
    // Object.keys(config).includes(req.params.provider)
    /facebook|twitter/.test(req.params.provider)
      ? next() : res.status(404).end()
  })
  .use(session({secret: 'grant', saveUninitialized: true, resave: true}))
  .use(grant(config))
  .get('/handle_facebook_callback', (req, res) => {
    res.end(JSON.stringify(req.query, null, 2))
  })
  .get('/handle_twitter_callback', (req, res) => {
    res.end(JSON.stringify(req.query, null, 2))
  })
  .listen(3000, () => console.log(`Express server listening on port ${3000}`))
```

**Comment by simov** [comment]
@Lordmau5, I've published v4.2.0 where the Dynamic Override feature is now disabled by defaults. So all you need to to is bump the version in your project. Let me know what do you think.

Also just FYI the Dynamic Override is now an [option](https://github.com/simov/grant#dynamic-override).

Thanks for the feedback!
